server/common/config/dataset_config.py

```diff
@@ -0,0 +1,214 @@
+import os
+from os.path import splitext, isdir
+
+from server.common.annotations.hosted_tiledb import AnnotationsHostedTileDB
+from server.common.annotations.local_file_csv import AnnotationsLocalFile
+from server.common.config.base_config import BaseConfig
+from server.common.errors import ConfigurationError, OntologyLoadFailure
+from server.compute.scanpy import get_scanpy_module
+from server.data_common.matrix_loader import MatrixDataLoader, MatrixDataType
+from server.db.db_utils import DbUtils
+
+
+class DatasetConfig(BaseConfig):
+    """Manages the config attribute associated with a dataset."""
+
+    def __init__(self, tag, app_config, default_config):
```

Comment:
Can you add types here? So something like `def __init__(self, tag: str, app_config: AppConfig, default_config: BaseConfig):`

(I'm just guessing at what these might be..)

Reply:
Because AppConfig creates the DatasetConfig/ServerConfig objects adding types seems to lead to circular import errors (even when done locally)